nevergrad/optimization/optimizerlib.py

```diff
@@ -1557,6 +1565,7 @@ def __init__(
     ) -> None:
         super().__init__(parametrization, budget=budget, num_workers=num_workers)
         assert budget is not None
+        self.restricted_budget = True
```

Comment:
don't add it when it is already added by the base class.

Reply:
this is also true for `budget_before_choosing` we should not be copied everywhere.

Reply:
Done x2 :-)